File: Backend/src/main/java/com/example/backend/entity/GroceryItemEntity.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Entity
@Table(name = "grocery_items")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class GroceryItemEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "grocery_id", nullable = false)
    private GroceryEntity grocery;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false, length = 20)
    private String unit;
}

File: MyApplication/data/src/main/java/com/example/data/Recipes/model/GroceryItemEntity.kt
package com.example.data.Recipes.model

import com.example.domain.Recipes.model.GroceryItem

data class GroceryItemEntity(
    val id: Int,
    val groceryId: Int,
    val name: String,
    val defaultUnit: String
) {
    fun toDomain(): GroceryItem {
        return GroceryItem(
            id = id,
            groceryId = groceryId,
            name = name,
            defaultUnit = defaultUnit
        )
    }
}